Document Hash-Related Image URL Structures

abuneri opened the pull request:
Wasn't exactly sure where the best place to put this is, chose to put it by the message formatting section since this is generally related to structure of discord client-specific things.

I realize this probably isnt the best design, so wouldn't mind suggestions about: 
1. the description at the top could use some work (if even needed at all)
2. http(s):// should probably be taken out of the table values? probably dont want this to render as a link, but don't want to confuse people into thinking they don't need it. not sure 
3. general table design

People ask about these url's in the discord api server all the time, so would be nice to have it documented.

Commit:
document the various image urls used by the client

SinisterRectus commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
+|---------|-----------------|
+| Custom Emoji | https://discordapp.com/api/emojis/[{emoji.id}](#DOCS_GUILD/emoji-object).png |
+| Guild Icon | https://cdn.discordapp.com/icons/[{guild.id}](#DOCS_GUILD/guild-object)/[{guild.icon}](#DOCS_GUILD/guild-object).[{format}](#DOCS_CHANNEL/image-formats)?size=[{size}](#DOCS_CHANNEL/image-sizes) |
+| Invite Splash | https://discordapp.com/splashes/[{guild.id}](#DOCS_GUILD/guild-object)/[{guild.splash}](#DOCS_GUILD/guild-object).jpg?size=2048 |
```
Splashes are not limited to invites, they are guild properties. Also missing `cdn.`

devsnek replied:
`https://cdn.discordapp.com/splashes/{guild_id}/{hash}.{format}`

devsnek commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
+
+| Type | Structure |
+|---------|-----------------|
+| Custom Emoji | https://discordapp.com/api/emojis/[{emoji.id}](#DOCS_GUILD/emoji-object).png |
```
`https://cdn.discordapp.com/emojis/{id}.png`

Commit:
fix splash naming

Commit:
add cdns and splash image formats

Commit:
link to section in relevant fields

i think doing something like this is useful as well, so when people look at these fields they know that an image url can be constructed from them

Commit:
fix cdn url for emojis

msciotti commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
 
 Using the markdown for either users, roles or channels will mention the target(s) accordingly.
 
+## URL Formatting
```
Actually, I think the entire `## Message Formatting` section should be moved to the `Reference.md` page along with this section. It's general information and not specific to channel.

msciotti commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
+| Custom Emoji | https://cdn.discordapp.com/emojis/[{emoji.id}](#DOCS_GUILD/emoji-object).png |
+| Guild Icon | https://cdn.discordapp.com/icons/[{guild.id}](#DOCS_GUILD/guild-object)/[{guild.icon}](#DOCS_GUILD/guild-object).[{format}](#DOCS_CHANNEL/image-formats)?size=[{size}](#DOCS_CHANNEL/image-sizes) |
+| Guild Splash | https://cdn.discordapp.com/splashes/[{guild.id}](#DOCS_GUILD/guild-object)/[{guild.splash}](#DOCS_GUILD/guild-object).[{format}](#DOCS_CHANNEL/image-formats)?size=2048 |
+| Invite Code | http://discord.gg/[{invite.code}](#DOCS_INVITE/invite-structure) |
```
Invite code doesn't really belong here. These are all images

msciotti commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
+
+###### Urls
+
+| Type | Structure |
```
I don't like "structure". Maybe `| Type | URL | ` instead, and name the sub-section `CDN Endpoints`

msciotti commented on docs/resources/Guild.md:
```diff
@@ -10,8 +10,8 @@ Guilds in Discord represent an isolated collection of users and channels, and ar
 | name | string | guild name (2-100 characters) |
-| icon | string | icon hash |
-| splash | string | splash hash |
+| icon | string | icon hash, [url format](#DOCS_CHANNEL/url-formatting) |
```
Rather than adding more text, just link `icon hash` and `splash hash` etc to the docs section.

msciotti reviewed (changes requested):
Solid info, just needs to be restructured a bit

matmen commented on docs/resources/Channel.md:
```diff
@@ -332,6 +332,34 @@ Discord utilizes a subset of markdown for rendering message content on its clien
+
+###### Image Sizes
+
+Power of 2 between 16 and 1024.
```
Wasn't this changed to a max size of 2048?

abuneri replied:
For User Avatar and Guild Icon, the ones who can have these "variables" sizes, 1024 is the maximum that they increase in size from my tests.

Commit:
changes from review

Commit:
update image format and size links

msciotti commented on docs/Reference.md:
```diff
@@ -116,3 +116,46 @@ The HTTP API implements a process for limiting and preventing excessive requests
+| Role | `<@&ROLE_SNOWFLAKE_ID>` | `<@&165511591545143296>` |
+| Custom Emoji | `<:NAME:ID>` | `<:mmLol:216154654256398347>` |
+
+Using the markdown for either users, roles or channels will mention the target(s) accordingly.
```
Should be `users, roles, or channels`. smh oxford comma deniers

msciotti commented on docs/Reference.md:
```diff
@@ -116,3 +116,46 @@ The HTTP API implements a process for limiting and preventing excessive requests
+
+## URL Formatting
+
+Discord utilizes the id's and hashes from specific objects to provide the images rendered in the client. This functionality uses the following formats:
```
Let's clarify this a little:

"Discord uses ids and hashes to render images in the client. These hashes can be retrieved through various API requests, like [Get User](#DOCS_USER/get-user). Below are the formats, size limitations, and CDN endpoints for images in Discord:"

msciotti commented on docs/Reference.md:
```diff
@@ -116,3 +116,46 @@ The HTTP API implements a process for limiting and preventing excessive requests
+
+Using the markdown for either users, roles or channels will mention the target(s) accordingly.
+
+## URL Formatting
```
This should really be `## Image Formatting`

msciotti reviewed (changes requested):
Few more clarifications and nits

Commit:
clairfy wording

Commit:
update field links
